lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -45,9 +57,98 @@
     seed_milestone_relationship_types,
 )
 
+from lms.djangoapps.ccx.models import CustomCourseForEdX
+
 # pylint: disable=protected-access
 
 
+class CoachAccessTestCaseCCX(SharedModuleStoreTestCase, LoginEnrollmentTestCase):
+    """
+    Test if user is coach on ccx.
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Set up course for tests
+        """
+        super(CoachAccessTestCaseCCX, cls).setUpClass()
+        cls.course = CourseFactory.create()
+
+    def setUp(self):
+        """
+        Set up tests
+        """
+        super(CoachAccessTestCaseCCX, self).setUp()
+
+        # Create ccx coach account
+        self.coach = AdminFactory.create()
+        self.client.login(username=self.coach.username, password="test")
+
+        # assign role to coach
+        role = CourseCcxCoachRole(self.course.id)
+        role.add_users(self.coach)
```

Comment:
I think you're correct, but it's *really* unclear that `add_users()` would save the role. In particular, that method doesn't even call `self.save()` -- instead, it creates new Django models, and saves *those* models to the database. The only reason that the role is saved is because those new Django models reference the role, and so in order for the new models to be saved, the `role` must also be implicitly saved.

Could you add an explicit `role.save()` line to this `setUp()` method? It will make it a *lot* clearer that this role gets saved. If I was confused, other people may be confused as well, and explicit is better than implicit.